chain/client/src/client.rs

```diff
@@ -394,7 +394,9 @@ impl Client {
         block: &Block,
     ) -> Result<(), Error> {
         let epoch_id = self.epoch_manager.get_epoch_id(block.hash())?;
-        let shard_layout = self.epoch_manager.get_shard_layout(&epoch_id)?;
+        let protocol_version = self.epoch_manager.get_epoch_protocol_version(&epoch_id)?;
+        let shard_layout =
+            self.epoch_manager.get_shard_layout_from_protocol_version(protocol_version);
```

Comment:
Why this change? It's the same but longer, so worse ;)

Reply:
Also in the future the shard layout may no longer be tied to protocol version (dynamic resharding). That of course is no argument given how hypothetical it is :)

Reply:
ah I think this is some code and this can be reverted.

Reply:
fixed.